gateway/mw_granular_access.go

```diff
@@ -34,33 +38,96 @@ func (m *GranularAccessMiddleware) ProcessRequest(w http.ResponseWriter, r *http
 		return nil, http.StatusOK
 	}
 
-	urlPath := m.Spec.StripListenPath(r.URL.Path)
+	gwConfig := m.Gw.GetConfig()
+
+	// Hook per-api settings here (m.Spec...)
```

Comment:
shall we add this part (+api level config override) while loading API (as new fields in API spec) instead of doing it  in middleware, so that it's done centrally and doesn't need to be duplicated

Reply:
we may need to revisit when doing perapi; nothing to action right now